app/containers/Forms/FuelRowIdField.tsx

```diff
@@ -17,26 +18,63 @@ export const FuelRowIdFieldComponent: React.FunctionComponent<Props> = (
    * Injects the list of Fuels in the schema, and remove `query` from the props
    * Other props are passed as-is to the StringField
    */
-  const fieldProps = useMemo(
+
+  const fuels = useMemo(
     () => ({
+      activeRowIds: props.query.activeFuels.edges.map(({node}) => node.rowId),
+      activeNames: props.query.activeFuels.edges.map(({node}) => node.name),
+
+      // Object containing the archived fuels, keyed on the rowId
+      archived: props.query.archivedFuels.edges.reduce(
+        (archivedDict, {node}) => {
+          archivedDict[node.rowId] = node.name;
+          return archivedDict;
+        },
+        {}
+      )
+    }),
+    [props.query]
+  );
+
+  const fieldProps = useMemo(() => {
+    return {
       ...props,
       schema: {
         ...props.schema,
-        enum: props.query.allFuels.edges.map(({node}) => node.rowId),
-        enumNames: props.query.allFuels.edges.map(({node}) => node.name)
+        enum: fuels.activeRowIds,
+        enumNames: fuels.activeNames
       },
       query: undefined
-    }),
-    [props]
-  );
+    };
+  }, [props]);
+
+  if (
+    fuels.activeRowIds.includes(props.formData) ||
+    props.formData === undefined
+  )
+    return <props.registry.fields.StringField {...fieldProps} />;
 
-  return <props.registry.fields.StringField {...fieldProps} />;
+  return (
+    <InputGroup>
+      <InputGroup.Text className="col-md-12">
+        {fuels.archived[props.formData]}
+      </InputGroup.Text>
+    </InputGroup>
+  );
 };
 
 export default createFragmentContainer(FuelRowIdFieldComponent, {
   query: graphql`
     fragment FuelRowIdField_query on Query {
-      allFuels(condition: {state: "active"}) {
+      activeFuels: allFuels(condition: {state: "active"}) {
```

Comment:
going forward do we want to use `filter` rather than `condition` for consistency? (This is a question, not necessarily a requested change). I've been replacing condition with filter when refactoring things.
Also: 
```suggestion
      activeFuels: allFuels(condition: {state: "active"} orderBy: NAME_ASC) {
```

Reply:
Happy to indulge! I doubt there is any performance concern between using the postgraphile filter plugin and the default graphQL syntax